db/models.py

```diff
@@ -50,3 +57,90 @@ class MovieSession(models.Model):
 
     def __str__(self) -> str:
         return f"{self.movie.title} {str(self.show_time)}"
+
+
+class Order(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    user = models.ForeignKey(
+        to=settings.AUTH_USER_MODEL,
+        on_delete=models.CASCADE,
+        related_name="orders"
+    )
+
+    class Meta:
+        ordering = ["-created_at"]
+
+    def __str__(self) -> str:
+        return str(self.created_at)
+
+
+class Ticket(models.Model):
+    movie_session = models.ForeignKey(
+        to=MovieSession,
+        on_delete=models.CASCADE,
+        related_name="tickets"
+    )
+    order = models.ForeignKey(
+        to=Order,
+        on_delete=models.CASCADE,
+        related_name="tickets"
+    )
+    row = models.IntegerField()
+    seat = models.IntegerField()
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                fields=["row", "seat", "movie_session"],
+                name="unique_ticket"
+            )
+        ]
+
+    def clean(self) -> None:
+        if not (
+            1 <= self.row <= self.movie_session.cinema_hall.rows
+        ):
+            raise ValidationError({
+                "row": [
+                    "row number must be in available range: "
+                    "(1, rows): "
+                    f"(1, {self.movie_session.cinema_hall.rows})"
+                ]
+            })
+
+        if not (
+            1 <= self.seat <= self.movie_session.cinema_hall.seats_in_row
+        ):
+            raise ValidationError({
+                "seat": [
+                    "seat number must be in available range: "
+                    "(1, seats_in_row): "
+                    f"(1, {self.movie_session.cinema_hall.seats_in_row})"
+                ]
+            })
+
+    def save(
+        self,
+        force_insert: bool = False,
+        force_update: bool = False,
+        using: str = None,
+        update_fields: list = None
+    ) -> None:
+        self.full_clean()
+        super(Ticket, self).save(
```

Comment:
After the last updates, there is no need to pass Ticket, self to super(). It will take them automatically